src/main/java/net/fabricmc/loom/providers/MinecraftAssetsProvider.java

```diff
@@ -76,18 +80,18 @@ public static void provide(MinecraftProvider minecraftProvider, Project project)
 			}
 		}
 
-		ProgressLogger progressLogger = ProgressLogger.getProgressFactory(project, MinecraftAssetsProvider.class.getName());
-		progressLogger.start("Downloading assets...", "assets");
+		project.getLogger().lifecycle(":downloading assets...");
+
+		Stack<ProgressLogger> loggers = new Stack<>();
```

Comment:
Deque please

Reply:
Why, whats the issue with stack?

Reply:
Stack is deprecated in newer versions of Java and will be removed eventually. Deque is the preferred replacement.

Reply:
ah, didnt know that. I will sort that out later. Thanks

Reply:
`Stack` (and all its other pre-collections friends) are obsolete but [not deprecated](http://hg.openjdk.java.net/jdk/jdk/file/e4fc1e578b3a/src/java.base/share/classes/java/util/Stack.java). Java still tries for backwards compatibility where possible so I wouldn't ever see them being removed given they've lasted the best part of 22 years not even being deprecated.

Also if you're using it for the thread safety you'll have to be careful the collection you replace it with is safe too (which `Deque` as an interface doesn't guarantee for example).

Reply:
moved to a `ConcurrentLinkedDeque`, will test once im at home.